packages/api/src/beacon/routes/node.ts

```diff
@@ -25,7 +25,7 @@ export const NetworkIdentityType = new ContainerType(
     p2pAddresses: ArrayOf(stringType),
     discoveryAddresses: ArrayOf(stringType),
     /** Based on Ethereum Consensus [Metadata object](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/p2p-interface.md#metadata) */
-    metadata: ssz.altair.Metadata,
+    metadata: ssz.fulu.Metadata,
```

Comment:
this is a breaking change in a sense that if the beacon node does not return `custody_group_count` field we will throw an error... the problem here is the api design, usually we want a new version of the api if we add new fields, I hinted at this here https://github.com/ethereum/beacon-APIs/pull/489 but since we did the same for altair I guess we are ok, I doubt anyone is using our api client to query this api and since we are merging this into peerDAS branch only for now I am fine with this, but let's keep an eye on this before we merge into unstable, ideally at this point all bns should return the field

Reply:
looks like this is already an [issue in our sim tests](https://github.com/ChainSafe/lodestar/actions/runs/14673095092/job/41202547677?pr=7751#step:6:88) because the Lighthouse version we run there doesn't support it

Reply:
Is there a better way to differentiate between the beacon-api types and the consensus-spec types when parsing them in those tests? It seems like the beacon API types are subtly different in that they don't require `syncnets` and `custody_group_count`, unlike the consensus-spec. I agree though that'd it'd definitely make it easier to match the types in both places

Reply:
so the problem is that there is no ssz type to represent an optiona field, even `Optional` in ssz means `<value> | null`, so if the field is omitted we always will have an issue

the alternative is to use `ssz.altair.Metadata` and handle `custody_group_count` outside of the ssz container, it's a bit hacky but I think we did this before somewhere, can try to find that

we usually try to use ssz containers but since the api can't support it anyways the hacky alternative would be fine

Reply:
I added a workaround, it's not pretty but should do the job for now